tests/core/tree_hash/test_containers.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+
+from ssz.sedes import (
+    List,
+    Serializable,
+    bytes_sedes,
+    uint32,
+    uint32_list,
+)
+from ssz.tree_hash.hash_eth2 import (
+    hash_eth2,
+)
+from ssz.tree_hash.tree_hash import (
+    hash_tree_root,
+)
+
+
+class SSZType1(Serializable):
+    fields = [
+        ('field1', uint32),
+        ('field2', bytes_sedes),
+        ('field3', uint32_list),
+    ]
+
+
+class SSZType2(Serializable):
+    fields = [
+        ('field2_1', SSZType1),
+        ('field2_2', List(SSZType1)),
+    ]
+
+
+class SSZType3(Serializable):
+    fields = [
+        ('field1', uint32),
+        ('field2', uint32),
+        ('field3', uint32),
+    ]
+
+    def __init__(self, field2, field1, field3, **kwargs):
+        super().__init__(field1=field1, field2=field2, field3=field3, **kwargs)
+
+
+class SSZType5(Serializable):
```

Comment:
It seems another special case along with `SSZUndeclaredFieldsType`. What's different between it and `SSZUndeclaredFieldsType`? What do you think about giving it a meaningful name for describing this case?

Reply:
Sorry, I tried giving TypeX some meaningful names, but then the code becomes lengthy and variables become indistinguishable.

Reply:
I also removed type5 and type6 as these test cases are less meaningful to tree hash.